core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -167,16 +172,25 @@ Partition get(StructLike key) {
     private final StructLike key;
     private long recordCount;
     private int fileCount;
+    private int deleteFileCount;
 
     Partition(StructLike key) {
       this.key = key;
       this.recordCount = 0;
       this.fileCount = 0;
+      this.deleteFileCount = 0;
     }
 
     void update(DataFile file) {
       this.recordCount += file.recordCount();
       this.fileCount += 1;
     }
+
+    void updateDeleteFile(List<DeleteFile> deleteFiles) {
```

Comment:
I think I would prefer to do this a little differently. Instead of updating the caller and passing both `task.file()` and `task.deletes()` separately, why not combine these into `update(FileScanTask)` and update both the file and deletes at once?

Reply:
Yes, if there is a metadata table that can load delete files, I think this is reasonable.

Reply:
Yes, I think it is reasonable.